mms/model_service_worker.py

```diff
@@ -453,11 +454,16 @@ def run_server(self):
             try:
                 log_msg("Waiting for a connection")
 
+                self.sock.settimeout(SOCKET_ACCEPT_TIMEOUT)
                 (cl_socket, _) = self.sock.accept()
+                self.sock.settimeout(socket.getdefaulttimeout())
```

Comment:
Why this is needed?

Reply:
Otherwise, it sets a 30s timeout for all socket operations which I believe includes listening and writing

Reply:
self.sock and cl_socket are different sockets. You would get a new socket after you accept a connection. self.sock is only used for listening and accepting new connections. Do you mean cl_socket is inheriting this property from the self.sock?

Reply:
The documentation does not say that it will inherit the timeout, so cl_socket should be fine